src/library/documentation.cpp

```diff
@@ -170,7 +180,8 @@ environment add_module_doc_string(environment const & env, std::string doc) {
     doc = process_doc(doc);
     auto ext = get_extension(env);
     ext.m_module_doc = cons(doc_entry(doc), ext.m_module_doc);
-    return update(env, ext);
+    auto new_env = update(env, ext);
+    return module::add(new_env, std::make_shared<doc_modification>(doc));
```

Comment:
What exactly does this do? Does it add stuff to .olean files? It's necessary to have this in order to preserve the modification - otherwise the changes seem lost when accessed in `lean.cpp`.